heartbeat/service_test.go

```diff
@@ -249,8 +255,8 @@ func TestHeartbeatOnlyWhenChallenged(t *testing.T) {
 	acct := ledgercore.AccountData{}
 
 	a.NoError(ledger.addBlock(table{joe: acct}))
-	ledger.waitFor(s, a)
-	a.Empty(sink.txns)
+	time.Sleep(time.Second)
+	a.True(sink.empty())
```

Comment:
how about eventually empty? It integrates yield/sleep and might finish earlier than hardcoded 1s sleep?